roaster-tm4j/roaster-tm4j-common/src/main/java/hu/icellmobilsoft/roaster/tm4j/common/DefaultTm4jReporter.java

```diff
@@ -0,0 +1,178 @@
+                .forEach(testCaseKey -> {
+                    Execution execution = createExecution(testCaseData, testCaseKey);
+                    execution.setStatus("Fail");
+                    execution.setComment(
+                            createCommentBase(testCaseData.getId()) +
+                                    createFailureComment(cause)
+                    );
+                    publishResult(execution);
+                });
+    }
+
+    @Override
+    public void reportDisabled(TestCaseData testCaseData, Optional<String> reason) {
+        getTestCaseIds(testCaseData.getTestMethod())
+                .forEach(testCaseKey -> {
+                    Execution execution = createExecution(testCaseData, testCaseKey);
+                    execution.setStatus("Blocked");
+                    execution.setComment(
+                            createCommentBase(testCaseData.getId()) +
+                                    createDisabledTestComment(reason)
+                    );
+                    publishResult(execution);
+                });
+    }
+
+    private void publishResult(Execution execution) {
+        tm4JService.postResult(config.getTestCycleKey(), execution);
+        log.info("Test result published to TM4J: [{0}]", execution.getTestCaseKey());
+    }
+
+    private Stream<String> getTestCaseIds(Method testMethod) {
+        return Arrays.stream(testMethod.getAnnotationsByType(TestCaseId.class))
+                .map(TestCaseId::value)
+                .map(testCaseId -> {
+                    if (tm4JService.isTestCaseExist(testCaseId)) {
+                        return testCaseId;
+                    } else {
+                        log.warn("Test case ID not found: [{0}]", testCaseId);
+                        return null;
+                    }
+                })
+                .filter(Objects::nonNull);
+    }
+
+    private Execution createExecution(TestCaseData testCaseData, String testCaseKey) {
+        Execution execution = new Execution();
+        execution.setProjectKey(config.getProjectKey());
+        execution.setTestCaseKey(testCaseKey);
+        execution.setActualStartDate(testCaseData.getStartTime());
+        execution.setActualEndDate(testCaseData.getEndTime());
+        execution.setExecutionTime(getDurationInMillis(testCaseData));
+        return execution;
+    }
+
+    private long getDurationInMillis(TestCaseData testCaseData) {
+        return testCaseData.getStartTime().until(testCaseData.getEndTime(), ChronoUnit.MILLIS);
+    }
+
+    private String createCommentBase(String uniqueId) {
+        return "Environment: " + config.getEnvironment().toUpperCase() +
+                "</br></br>" +
+                "Test method: " + uniqueId;
+    }
+
+    private String createFailureComment(Throwable cause) {
+        return "</br></br>Reason of failure: " + htmlEscape(cause.toString());
+    }
+
+    private String htmlEscape(String string) {
+        return string
```

Comment:
erre leteznek tool-ok mint pl URLEncoder

Reply:
Nekem html encode kell, olyan class meg nincs a projektben. Ezért a pár sorért pedig nem húznék be egy egész lib-et.

Reply:
commons-text a roaster-api -ban benne van, StringEscapeUtils mehet nyugodtan